flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistryImpl.java

```diff
@@ -167,7 +152,6 @@ public StreamStateHandle registerReference(
             }
         } // end of synchronized (registeredStates)
 
-        scheduleAsyncDelete(scheduledStateDeletion);
```

Comment:
Can you please expand the commit message to clarify why this is not necessary anymore?

And probably rename the commit from 
`[hotfix][state-changelog] not trigger new materialization until previous one confirmed or failed or cancelled`
to 
`[hotfix][state-changelog] Don't trigger new materialization unless the previous one is confirmed/failed/cancelled`